src/components/SignUp/MyPic.js

```diff
@@ -27,60 +27,64 @@ class MyPic extends Component {
   }
 
   componentWillReceiveProps(props) {
-    this.setState({
-      isDefault: false,
-      uri: props.uri,
-    });
-  }
-
-  render() {
-    let showPicker = () => this.showPicker();
-    let source = this.state.source;
-    if (!this.state.isDefault && this.state.uri !== undefined) {
-      source = { uri: this.state.uri };
+    if (!this.state.uri) {
```

Comment:
Not sure why you set up with this.state instead of setState in componentWillReceiveProps cycle.